flang/include/flang/Optimizer/Transforms/Passes.td

```diff
@@ -252,6 +252,16 @@ def MemoryAllocationOpt : Pass<"memory-allocation-opt", "mlir::func::FuncOp"> {
   ];
 }
 
+// This needs to be a "mlir::ModuleOp" pass, because it inserts global constants
+def ConstantArgumentGlobalisationOpt : Pass<"constant-argument-globalisation-opt", "mlir::ModuleOp"> {
+  let summary = "Convert constant function arguments to global constants.";
+  let description = [{
+    Convert scalar literals of function arguments to global constants.
+  }];
+  let dependentDialects = [ "fir::FIROpsDialect" ];
+  let constructor = "::fir::createConstantArgumentGlobalisationPass()";
```

Comment:
If you remove this line, tablegen will create a constructor for you. It will be called `createConstantArgumentGlobalisationOpt`.